src/components/GlobalPreferences/Notifications/ManageNotifications.js

```diff
@@ -0,0 +1,384 @@
+import React, { useState, useEffect, useCallback } from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import { AppType } from '../../../prop-types'
+import {
+  Box,
+  Button,
+  GU,
+  DataView,
+  IconTrash,
+  IconCheck,
+  IconMail,
+  Info,
+  LoadingRing,
+  breakpoint,
+  font,
+  useTheme,
+  textStyle,
+  Split,
+} from '@aragon/ui'
+import LocalIdentityBadge from '../../../components/IdentityBadge/LocalIdentityBadge'
+import {
+  getSubscriptions,
+  deleteAccount,
+  deleteSubscriptions,
+} from './notification-service-api'
+import {
+  NOTIFICATION_SERVICE_TOKEN_KEY,
+  NOTIFICATION_SERVICE_EMAIL_KEY,
+} from './constants'
+import { SubscriptionsForm } from './SubscriptionsForm'
+import SubscriptionFilters from './SubscriptionFilters'
+
+export default function ManageNotifications({
+  apps,
+  dao,
+  email,
+  onLogout,
+  token,
+}) {
+  const [apiError, setApiError] = useState(null)
+  const [isFetching, setIsFetching] = useState(true)
+  const [subscriptions, setSubscriptions] = useState([])
+
+  const fetchSubscriptions = useCallback(() => {
+    getSubscriptions(token)
+      .then(subscriptions => {
+        setSubscriptions(subscriptions)
+        setIsFetching(false)
+        return subscriptions
+      })
+      .catch(error => {
+        setIsFetching(false)
+        setApiError(error.message)
+      })
+  }, [token])
+
+  useEffect(() => {
+    if (!token) return
+    fetchSubscriptions()
+  }, [fetchSubscriptions, token])
+
+  return (
+    <React.Fragment>
+      {apiError && (
+        <Info
+          css={`
+            margin-bottom: ${GU}px;
+          `}
+          mode="error"
+        >
+          {apiError}
+        </Info>
+      )}
+      <Split
+        primary={
+          <SubscriptionsForm
+            onApiError={setApiError}
+            onCreate={fetchSubscriptions}
+            dao={dao}
+            apps={apps}
+            token={token}
+            isFetchingSubscriptions={isFetching}
+            subscriptions={subscriptions}
+          />
+        }
+        secondary={
+          <React.Fragment>
+            <Box heading="Signed In With Email">
+              {email}
+              <Button
+                css={`
+                  margin-top: ${2 * GU}px;
+                `}
+                wide
+                onClick={onLogout}
+              >
+                Sign Out
+              </Button>
+            </Box>
+            <DeleteAccount
+              onApiError={setApiError}
+              token={token}
+              onLogout={onLogout}
+            />
+          </React.Fragment>
+        }
+      />
+      {!isFetching && subscriptions.length > 0 && (
```

Comment:
Maybe provide a `LoadingRing` while `isFetching` is true? This is currently what gets rendered while `getSubscriptions` has not resolved:

![image](https://user-images.githubusercontent.com/3072458/63693502-881cca00-c814-11e9-8e87-e733466e1684.png)

And it doesn't really say there is something happening.